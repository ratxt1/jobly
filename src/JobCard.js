import React, { Component } from 'react';

class JobCard extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(evt) {
    this.props.apply(evt.target.id);
  }

  render() {
    let { id, title, salary, equity, state } = this.props;
    return (
      <div className="card mx-auto mb-1">
        <div className="card-body">
          <h5 className="card-title">{title}</h5>
          <p className="card-text">Salary: {salary}</p>
          <p className="card-text">Equity: {equity}</p>
          <button
            id={id}
            className="btn btn-danger"
            disabled={state}
            onClick={this.handleClick}>
            { (!state) ? "APPLY" : "APPLIED"}
          </button>
        </div>
      </div>
    )
  }
}

export default JobCard;


