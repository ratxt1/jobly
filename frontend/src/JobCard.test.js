import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import JobCard from "./JobCard";

let wrapper;

let j = {
    id: 1,
    title: "test-job",
    salary: 100, 
    equity: 0.5
};

it("renders without crashing", function() {
    shallow(<JobCard key={j.id} {...j}/>);
  });
  
// snapshot test
it("matches snapshot", function() {
    wrapper = shallow(<JobCard key={j.id} {...j}/>);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

