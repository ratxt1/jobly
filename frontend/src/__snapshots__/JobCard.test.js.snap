// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card mx-auto mb-1"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      test-job
    </h5>
    <p
      className="card-text"
    >
      Salary: 
      100
    </p>
    <p
      className="card-text"
    >
      Equity: 
      0.5
    </p>
  </div>
</div>
`;
